src/h5web/visualizations/containers/hooks.ts

```diff
@@ -17,3 +27,47 @@ export function useDatasetValues(
 
   return valuesReader();
 }
+
+export function useNxAxisParams(
+  nxDataGroup: HDF5Group,
+  metadata: HDF5Metadata,
+  signalName: string,
+  axesNames: (string | undefined)[]
+): Record<string, AxisParams> | undefined {
+  const nxNames = [signalName, ...axesNames];
+
+  const nxDatasets = nxNames.reduce<Record<string, HDF5Dataset>>(
+    (acc, axis) => {
+      if (!axis) {
+        return acc;
+      }
+
+      const dataset = getLinkedEntity(nxDataGroup, metadata, axis);
+      if (dataset && isDataset(dataset)) {
+        acc[axis] = dataset;
+      }
+      return acc;
+    },
+    {}
+  );
+
+  const dsetValues = useDatasetValues(
+    Object.fromEntries(
+      Object.entries(nxDatasets).map(([axis, dataset]) => [axis, dataset.id])
+    )
+  );
+
+  if (!dsetValues) {
+    return undefined;
+  }
+
+  return Object.entries(nxDatasets).reduce<Record<string, AxisParams>>(
+    (acc, [axis, dataset]) => {
+      const values = dsetValues[axis];
+      assertArray<number>(values);
+      acc[axis] = { values, label: getDatasetLabel(dataset, axis) };
+      return acc;
+    },
+    {}
+  );
+}
```

Comment:
The new hook that fetches values and labels for axes and the signal altogether. The aim is to get `AxisParams` all the way down to the lower-level Vis.